src/senselab/audio/tasks/forced_alignment/align.py

```diff
@@ -0,0 +1,587 @@
+                    word_score = round(word_chars["score"].mean(), 3)
+
+                    word_segment = SingleWordSegment(word=word_text, start=word_start, end=word_end, score=word_score)
+
+                    sentence_words.append(word_segment)
+                    word_segments.append(word_segment)
+
+                aligned_subsegment = SingleAlignedSegment(
+                    text=sentence_text, start=sentence_start, end=sentence_end, words=sentence_words, chars=word_chars
+                )
+                aligned_subsegments.append(aligned_subsegment)
+
+                if return_char_alignments:
+                    curr_chars = curr_chars[["char", "start", "end", "score"]]
+                    curr_chars.fillna(-1, inplace=True)
+                    curr_chars = curr_chars.to_dict("records")
+                    curr_chars = [{key: val for key, val in char.items() if val != -1} for char in curr_chars]
+                    aligned_subsegments[-1]["chars"] = curr_chars
+
+            if aligned_subsegments:
+                aligned_subsegments_df = pd.DataFrame(aligned_subsegments)
+
+                aligned_subsegments_df["start"] = _interpolate_nans(
+                    aligned_subsegments_df["start"], method=interpolate_method
+                )
+                aligned_subsegments_df["end"] = _interpolate_nans(
+                    aligned_subsegments_df["end"], method=interpolate_method
+                )
+                agg_dict = {"text": " ".join, "words": "sum"}
+                if model_lang in LANGUAGES_WITHOUT_SPACES:
+                    agg_dict["text"] = "".join
+                if return_char_alignments:
+                    agg_dict["chars"] = "sum"
+                aligned_subsegments_df.groupby(["start", "end"], as_index=False).agg(agg_dict)
+                aligned_subsegments = aligned_subsegments_df.to_dict("records")
+
+        aligned_segments.extend(aligned_subsegments)
+
+    return (aligned_segments, word_segments)
+
+
+def convert_to_scriptline(data: AlignedTranscriptionResult) -> List[ScriptLine]:
+    """Convert a dictionary of segments and word segments to a list of ScriptLine objects.
+
+    Args:
+        data (AlignedTranscriptionResult): The input dictionary with segments and word segments.
+
+    Returns:
+        List[ScriptLine]: The list of ScriptLine objects.
+    """
+    segments = data["segments"]
+    script_lines = []
+
+    for segment in segments:
+        words = segment["words"]
+        word_chunks = [ScriptLine(text=word["word"]) for word in words]
+
+        # Handle 'nan' end values by setting them to None
+        start = segment["start"]
+        end: Optional[float] = segment["end"]
+        if end is not None and (isinstance(end, float) and math.isnan(end)):
+            end = None
+
+        script_line = ScriptLine(text=segment["text"], start=start, end=end, chunks=word_chunks)
+        script_lines.append(script_line)
+
+    return script_lines
+
+
+def align(
```

Comment:
Since you reviewed the voice activity detection and speaker diarization tasks, you know that they delivers a list of ScriptLine objects. And also the speech to text outputs a list of ScriptLine objects. See https://github.com/sensein/senselab/blob/main/src/senselab/utils/data_structures/script_line.py
Is there any specific reason why you don't use it here? I think it may be convenient using ScriptLine, so that when you will implement the WhisperX workflow your life is going to be easier (the output of one component can directly be passed as an input to the second one in the sequence)